### PR TITLE
SilverStripe 4 compatibility

### PR DESCRIPTION
**Pull request by robbieaverill**
Updated for compatibility with SilverStripe 4 in its current state. Brief summary of changes:

* Composer: dev alias added, PSR-4 autoloading compatibility
* Structure: "code" -> "src", namespacing added and implemented
* `sourceList` property with getter/setter added to circumvent change in ORM's `DataList::setSource` change
* Chosen JS: selector names updated ([thanks Ingo](https://github.com/silverstripe/silverstripe-asset-admin/issues/135#issuecomment-217264582))
* Travis: config updated for SS4 (PHP 7.x is no longer allowed to fail)
* PSR-2: some style formatting changes

---

## Requirements

- ~Branch current master to `2` before merge~
- ~Update target branch of this pull request to `2`~
- [x] Current versions branched off to their own branches in preparation for `master` becoming 4.x compatible

### CODE REVIEW

**Comment by dhensby** [comment]
Master is now aliased as `2.x-dev` - please update the `.travis.yml` and `composer.json` files

**Comment by robbieaverill** [comment]
I've rebased the branch to reflect @dhensby's latest alias change in master. Ready for review.

File: .travis.yml [hunk 1]
```diff
@@ -4,26 +4,18 @@ sudo: false
...
-env:
-  - DB=MYSQL CORE_RELEASE=3.2
-
 matrix:
   include:
+    - php: 5.5
+      env: DB=MYSQL CORE_RELEASE=master
     - php: 5.6
-      env: DB=MYSQL CORE_RELEASE=3
+      env: DB=MYSQL CORE_RELEASE=master
```
Comment (dhensby) [inline .travis.yml]: `CORE_RELEASE=4` may be better

Comment (robbieaverill) [inline .travis.yml]: Updated

**Comment by dhensby** [comment]
Ready?

**Comment by robbieaverill** [comment]
Not 100% tested but base functionality is working so far. I think it's fine to start with at least. It could use a bit of styling updated to match the new admin theme at some point.

**Pull request merged**